device-telemetry/app/com/microsoft/azure/iotsolutions/devicetelemetry/services/storage/cosmosDb/StorageClient.java

```diff
@@ -143,25 +145,20 @@ public Document deleteDocument(String databaseName, String colId, String docId)
     }
 
     @Override
-    public Status ping() {
-        URI response = null;
-        String name = "Storage";
-
-        if (this.client != null) {
-            response = this.client.getReadEndpoint();
-        }
-
-        if (response != null) {
-            return new Status(
-                name,
-                true,
-                "Storage alive and Well!");
-        } else {
-            return new Status(
-                name,
-                false,
-                "Could not reach storage service. Check connection string");
+    public StatusResultServiceModel ping() {
+        StatusResultServiceModel result = new StatusResultServiceModel(false, "Storage check failed");
+        DatabaseAccount response = null;
+        try {
+            if (this.client != null) {
```

Comment:
should there be a more specific error if the client was not set up properly?

Reply:
done